CorsixTH/Src/iso_fs.cpp

```diff
@@ -133,63 +283,34 @@ bool iso_filesystem::initialise(FILE* fRawFile)
 
     // The first 16 sectors are reserved for bootable media.
     // Volume descriptor records follow this, with one record per sector.
-    for(uint32_t iSector = 16; seek_to_sector(iSector); ++iSector)
-    {
-        uint8_t aBuffer[190];
-        if(!read_data(sizeof(aBuffer), aBuffer))
+    for(uint32_t iSector = first_filesystem_sector; seek_to_sector(iSector); ++iSector) {
+        uint8_t aBuffer[root_directory_offset + root_directory_entry_size];
+        if(!read_data(sizeof(aBuffer), aBuffer)) {
             break;
+        }
         // CD001 is a standard identifier, \x01 is a version number
-        if(std::memcmp(aBuffer + 1, "CD001\x01", 6) == 0)
-        {
-            if(aBuffer[0] == vdt_privary_volume)
-            {
-                sector_size = read_native_int<uint16_t>(aBuffer + 128);
-                find_hosp_directory(aBuffer + 156, 34, 0);
-                if(file_count == 0)
-                {
+        if(std::memcmp(aBuffer + 1, "CD001\x01", 6) == 0) {
+            if(aBuffer[0] == vdt_primary_volume) {
+                sector_size = bytes_to_uint16_le(aBuffer + sector_size_offset);
+                find_hosp_directory(aBuffer + root_directory_offset, root_directory_entry_size, 0);
+                if(files.empty()) {
                     set_error("Could not find Theme Hospital data directory.");
                     return false;
-                }
-                else
-                {
+                } else {
                     return true;
                 }
-            }
-            else if(aBuffer[0] == vdt_terminator)
+            } else if(aBuffer[0] == vdt_terminator) {
                 break;
+            }
         }
     }
     set_error("Could not find primary volume descriptor.");
     return false;
 }
 
-int iso_filesystem::filename_compare(const void* lhs, const void* rhs)
-{
-    return std::strcmp(
-        reinterpret_cast<const file_metadata*>(lhs)->path,
-        reinterpret_cast<const file_metadata*>(rhs)->path);
-}
-
-char iso_filesystem::normalise(char c)
-{
-    if(c == '_') // underscore to hyphen
-        return '-';
-    else if('a' <= c && c <= 'z') // ASCII lowercase to ASCII uppercase
-        return static_cast<char>(c - 'a' + 'A');
-    else
-        return c;
-}
-
-void iso_filesystem::trim_identifier_version(const uint8_t* sIdent, uint8_t& iLength)
+bool iso_filesystem::filename_compare(const file_metadata& lhs, const file_metadata& rhs)
```

Comment:
`compare` is a ambiguous, eg equality checking is also comparing. Perhaps `filename_sort_compare` or `filename_orderer` ?

Reply:
hmm, existing name, but the functionality was changed. Don't like confusing what it does with where it's used. otoh it's simple enough it could be a lambda. `file_path_less_than`?

Reply:
It parallels std::less_than